caput/memh5.py

```diff
@@ -2139,9 +2140,19 @@ def _map_unicode(value):
 
     def _map_json(value):
         # Serialize/deserialize "special" json values
+
+        # Datetimes often appear in the configs (as they are parsed by PyYAML),
+        # so we need to serialise them back to strings
+        def _convert_datetime(v):
+            if isinstance(v, datetime.datetime):
+                return v.isoformat()
+            raise TypeError("Can not JSON serialise object of type %s" % repr(type(v)))
```

Comment:
It looks like you are supposed to let the default default raise the TypeError: 
https://docs.python.org/3/library/json.html#json.JSONEncoder.default
```
# Let the base class default method raise the TypeError
   return json.JSONEncoder.default(self, o)
```

Reply:
It's not clear how to  do that if I'm not creating my own JSONEncoder instance. Thoughts?

Reply:
Hmm. Do you have a test for the code you added?

Reply:
Nothing clean. I've just been running through an example pipeline that fails.